react/Components/input.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useContext, FC } from 'react'
+import { Input } from 'vtex.styleguide'
+
+import Context from '../Context/context'
+
+interface Props {
+  name: string
+}
+const InputArea: FC<Props> = ({ name }: Props) => {
+  const provider = useContext(Context)
+
+  const updateValueName = (event: { target: { value?: string } }) => {
+    provider.setName(event.target.value ?? '')
+  }
+
+  const updateValueText = (event: { target: { value?: string } }) => {
+    provider.setText(event.target.value ?? '')
+  }
+  return (
+    <>
+      <div className="mb5 mt5">
+        <Input
+          name={name === 'name' ? 'name' : 'text'}
+          placeholder={
+            name === 'name' ? 'Escolha o nome da badges' : 'Insira seu texto'
+          }
+          size="large"
+          label={name === 'name' ? 'Nome da badge' : 'Insira o texto da badge'}
+          value={name === 'name' ? provider.name : provider.text}
+          onChange={name === 'name' ? updateValueName : updateValueText}
+        />
```

Comment:
Multiple tests to the same thing are very expensive, because the code runs the same `if` and can run just one time.

What do you think in refactor this to a conditional with two returns, like this:

```suggestion

const getInput = useCallback((inputType: inputType ) => {
if(inputType === 'name'){
return <Input 
                name={'name'} 
                placeholder={'Escolha o nome da badges'} 
                size="large"
                label={'Nome da badge'}
                value={provider.name}
                onChange={updateValueName}
              />
} else {
return <Input
                name={'text'}
                placeholder={'Insira seu texto'}
                size="large"
                label={'Insira o texto da badge'}
                value={provider.text}
                onChange={updateValueText}
              />
}
}, [...])

...

{getInput(name)}
```